backend/app/controllers/users_controller.rb

```
@@ -13,6 +14,44 @@ def show
+        end
+
+        render json: {
+            user_id: user.id,
+            email: user.email,
+            current_username: user_profile.username,
+            total_changes: formatted_history.length,
+            changes: formatted_history
+        }
+    rescue ActiveRecord::RecordNotFound
+        render json: { error: "User not found" }, status: :not_found
+    end
```

RuboCop (1.76.1)

[convention] Assignment Branch Condition size for `username_history` is too high. [<5, 23, 3> 23.73/23]

(Metrics/AbcSize)